source/command-monitoring/tests/unified/redacted-commands.yml

```diff
@@ -177,76 +210,131 @@ tests:
         object: *database
         arguments:
           commandName: hello
           command:
-            hello: "private"
-            speculativeAuthenticate: "foo"
-        expectError:
-          isError: true
+            hello: 1
+            speculativeAuthenticate:
+              saslStart: 1
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              commandName: hello
+              command:
+                hello: { $$exists: false }
+                speculativeAuthenticate: { $$exists: false }
+          - commandSucceededEvent:
+              commandName: hello
+              reply:
+                # We expect the reply not to be redacted, as the server can't
+                # successfully authenticate us, which means the reply does not
+                # need to be redacted
+                isWritablePrimary: { $$exists: true }
+                # This assertion will currently always hold true since we're
+                # not expecting successful authentication, in which case this
+                # field is missing anyways.
+                speculativeAuthenticate: { $$exists: false }
+
+  - description: "legacy hello with speculative authenticate"
+    operations:
       - name: runCommand
         object: *database
         arguments:
           commandName: ismaster
           command:
-            ismaster: "private"
-            speculativeAuthenticate: "foo"
-        expectError:
-          isError: true
+            ismaster: 1
+            speculativeAuthenticate:
+              saslStart: 1
       - name: runCommand
         object: *database
         arguments:
           commandName: isMaster
           command:
-            isMaster: "private"
-            speculativeAuthenticate: "foo"
-        expectError:
-          isError: true
+            isMaster: 1
+            speculativeAuthenticate:
+              saslStart: 1
     expectEvents:
       - client: *client
         events:
           - commandStartedEvent:
-              commandName: hello
-              command: { hello: { $$exists: false } }
-          - commandStartedEvent:
               commandName: ismaster
-              command: { ismaster: { $$exists: false } }
+              command:
+                ismaster: { $$exists: false }
+                speculativeAuthenticate: { $$exists: false }
+          - commandSucceededEvent:
+              commandName: ismaster
+              reply:
+                ismaster: { $$exists: true }
+                speculativeAuthenticate: { $$exists: false }
           - commandStartedEvent:
               commandName: isMaster
-              command: { isMaster: { $$exists: false } }
+              command:
+                isMaster: { $$exists: false }
+                speculativeAuthenticate: { $$exists: false }
+          - commandSucceededEvent:
+              commandName: isMaster
+              reply:
+                ismaster: { $$exists: true }
```

Comment:
Uppercase `isMaster` here?

Reply:
The response is always lowercased, regardless of the command being invoked (output truncated to show relevant fields only):
```
MongoDB Enterprise > db.runCommand({ismaster: 1})
{
	"ismaster" : true,
	"ok" : 1
}
MongoDB Enterprise > db.runCommand({isMaster: 1})
{
	"ismaster" : true,
	"ok" : 1
}
```